client/human/HumanClientFSM.cpp

```diff
@@ -114,9 +114,13 @@ boost::statechart::result WaitingForSPHostAck::react(const HostSPGame& msg) {
 
 boost::statechart::result WaitingForSPHostAck::react(const Disconnection& d) {
     if (TRACE_EXECUTION) DebugLogger() << "(HumanClientFSM) PlayingGame.Disconnection";
-    //Note: Any transit<> transition must occur before the MessageBox().
-    // MessageBox blocks and can allow other events to transit<> to a new state
-    // which makes this transit fatal.
+    // Note: All transitions, transit<> or disable_event(), must occur in the order
```

Comment:
I laughed when I read "appears tautological".

I expanded the comment so that it hopefully is clear why the incorrect code ordering crashes. 

The comment was long and repeated. I moved it to the top of the file, with doxygen notation to create a separate page and added references to it at all the locations in code where it was repeated.